src/meta/client/MetaClient.cpp

```diff
@@ -107,6 +110,35 @@ void MetaClient::stop() {
     isRunning_ = false;
 }
 
+void MetaClient::dnsQUeryThreadFunc() {
+    SCOPE_EXIT {
+        bgThread_->addDelayTask(
+            FLAGS_meta_domain_ttl_secs * 1000, &MetaClient::dnsQUeryThreadFunc, this);
+    };
+    // we need update addr from dns.
+    std::vector<network::InetAddress> addrs;
+    std::vector<int> addrsIndex;
+    for (auto it = addrs.cbegin(); it != addrs.cend(); ++it) {
+        if (it->isResolved()) {
+            try {
+                auto newAddr = network::InetAddress(it->getHostStr(), it->getPort(), true);
+                if (newAddr != (*it)) {
+                    addrs.emplace_back(newAddr);
+                    addrsIndex.emplace_back(std::distance(addrs.cbegin(), it));
+                }
+            } catch (std::exception& e) {
+                LOG(ERROR) << "update meta addr failed, err: " << e.what();
+            }
+            continue;
+        }
+    }
+    folly::SharedMutexReadPriority::WriteHolder holder(addrsLock_);
+    for (auto index : addrsIndex) {
+        LOG(INFO) << "meta will update host from " << addrs_[index] << " to: " << addrs[index];
+        addrs_[index] = std::move(addrs[index]);
```

Comment:
Is this correct? Can we make sure that dns return the `addrs` in the order when we create `MetaClient`? I suppose that the `addrs_` is loaded from configuration.

Reply:
Addrs_ will be used in updateActive and updateLeader to update the corresponding IP address.